src/pointer.cr

```diff
@@ -495,10 +495,18 @@ struct Pointer(T)
   # ptr.to_slice(6) # => Slice[0, 0, 0, 13, 14, 15]
   # ```
   def clear(count = 1)
-    Intrinsics.memset(self.as(Void*), 0_u8, (count * sizeof(T)).to_u32, 0_u32, false)
+    Intrinsics.memset(self.as(Void*), 0_u8, bytesize(count), 0_u32, false)
   end
 
   def clone
     self
   end
+
+  private def bytesize(count)
+    {% if flag?(:x86_64) %}
```

Comment:
Does this apply also to `aarch64`?

Reply:
Using `LibC::SizeT.new(count)` would avoid the flags altogether. Maybe introducing SizeT is unavoidable after all?

Reply:
Surely if the semantics of pointer address is to sleazy be 64bits, this method should also always use 64bits?

Reply:
I changed it to `flag?(:bits64)`. The 32 bits branch needs to check that the value isn't bigger than `UInt32::MAX`.